docs/getting-started.md

```diff
@@ -0,0 +1,276 @@
+```bash
+git clone https://github.com/firecracker-microvm/firecracker.git
+```
+Change into the new directory, and build with Cargo.  Make sure to enable the
+optional `vsock` feature using the `--features vsock` flag.
+
+> Note: Firecracker normally builds a statically-linked binary with musl libc.
+> On Amazon Linux 2, you must specify `--target x86_64-unknown-linux-gnu`
+> because musl libc is not available.  Switching to this target changes the set
+> of syscalls invoked by Firecracker.  If you intend to jail Firecracker using
+> seccomp, you must adjust your seccomp profile for these changes.
+
+```bash
+git checkout v0.12.0 # latest released tag
+cargo build --release --features vsock # --target x86_64-unknown-linux-gnu
+```
+
+### Download appropriate filesystem images
+
+You can use the following images for the kernel and a default user-space:
+
+```bash
+curl -fsSL -o hello-vmlinux.bin https://s3.amazonaws.com/spec.ccfc.min/img/hello/kernel/hello-vmlinux.bin
+curl -fsSL -o hello-rootfs.ext4 https://s3.amazonaws.com/spec.ccfc.min/img/hello/fsfiles/hello-rootfs.ext4
+```
+
+We will be modifying the `hello-rootfs.ext4` file in a subsequent step.
+
+### Install containerd
+
+You can install containerd from the
+[upstream binaries](https://github.com/containerd/containerd/releases/tag/v1.2.1),
+[build it yourself](https://github.com/containerd/containerd/blob/master/BUILDING.md),
+or use the containerd binary included with recent releases of Docker (like
+18.09).  We do not recommend using older versions of containerd.
+
+The [quickstart guide](quickstart.md) has example commands for building and
+installing containerd.
+
+### Install runc
+
+You can install runc from the
+[upstream binaries](https://github.com/opencontainers/runc/releases/tag/v1.0.0-rc6),
+[build it yourself](https://github.com/containerd/containerd/blob/v1.2.1/RUNC.md),
+or use the runc binary included with recent releases of Docker (like 18.06).  We
+do not recommend using older versions of containerd.
+
+Depending on whether or not the root filesystem of the microVM contains an
+installation of glibc, you may need to have a statically-linked build of runc.
+
+The [quickstart guide](quickstart.md) has example commands for building and
+installing runc, with static linking.
+
+
+### Clone this repository and build the components
+
+Clone this repository to your computer in a directory of your choice.
+We recommend choosing a directory outside of `$GOPATH` or `~/go`.
+
+```bash
+git clone https://github.com/firecracker-microvm/firecracker-containerd
+make STATIC_AGENT=true
+```
+
+Once you have built the runtime, be sure to place the
+`containerd-shim-aws-firecracker` binary on your `$PATH`.
+
+### Inject the agent, runc shim, and runc into the rootfs image
+
+firecracker-containerd needs three components to be embedded in the root
+filesystem image.  They are:
+
+* `runc` - used to run containers
+* `containerd-shim-runc-v1` - containerd's default shim for controlling runc
```

Comment:
Is this actually required inside the VM?

Reply:
Nope.  Thanks for catching this, I'll remove it.